sql/parser/eval.go

```diff
@@ -535,6 +728,15 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (Datum, error){
 	cmpArgs{EQ, floatType, floatType}: func(left Datum, right Datum) (Datum, error) {
 		return DBool(left.(DFloat) == right.(DFloat)), nil
 	},
+	cmpArgs{EQ, dateType, dateType}: func(left Datum, right Datum) (Datum, error) {
+		return DBool(left.(DDate).Equal(right.(DDate).Time)), nil
+	},
+	cmpArgs{EQ, timestampType, timestampType}: func(left Datum, right Datum) (Datum, error) {
```

Comment:
What we can do is drop time zone completely and have the user specify what time zone they would like to see. Thoughts?